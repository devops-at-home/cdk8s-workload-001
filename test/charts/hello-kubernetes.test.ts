import { App } from 'cdk8s';
import {
    HelloKubernetes,
    HelloKubernetesConfig,
    parseInputs,
} from '../../src/charts/hello-kubernetes.chart';

describe('Snapshot tests', () => {
    test('Hello Kubernetes', () => {
        const app = new App();

        const chart = new HelloKubernetes(app, 'HelloKubernetes');

        expect(chart).toMatchSnapshot();
    });
});

describe('Unit tests', () => {
    test('parseInputs function default values', () => {
        const response = parseInputs({});
        expect(response.containerPort).toStrictEqual(8080);
        expect(response.image).toStrictEqual('paulbouwer/hello-kubernetes:1.10.1');
        expect(response.name).toStrictEqual('hello-kubernetes');
        expect(response.sslRedirect).toStrictEqual(false);
        expect(response.host).toStrictEqual('test.h6020-001.devops-at-ho.me');
    });

    test('parseInputs function with inputs', () => {
        const inputs: HelloKubernetesConfig = {
            containerPort: 8081,
            image: 'some image',
            name: 'some name',
            sslRedirect: true,
            host: 'some host',
        };

        const response = parseInputs(inputs);
        expect(response.containerPort).toStrictEqual(inputs.containerPort);
        expect(response.image).toStrictEqual(inputs.image);
        expect(response.name).toStrictEqual(inputs.name);
        expect(response.sslRedirect).toStrictEqual(inputs.sslRedirect);
        expect(response.host).toStrictEqual(inputs.host);
    });
});
