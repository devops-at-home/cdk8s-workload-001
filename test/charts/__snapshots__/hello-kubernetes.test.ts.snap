// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Hello Kubernetes 1`] = `
HelloKubernetes {
  "_disableResourceNameHashes": false,
  "_labels": {},
  "namespace": undefined,
  "node": Node {
    "_children": {
      "deployment": KubeDeployment {
        "apiGroup": "apps",
        "apiVersion": "apps/v1",
        "chart": [Circular],
        "kind": "Deployment",
        "metadata": ApiObjectMetadataDefinition {
          "_additionalAttributes": {
            "labels": {
              "app.kubernetes.io/name": "hello-kubernetes",
            },
            "name": "hello-kubernetes",
            "namespace": undefined,
          },
          "annotations": {},
          "finalizers": [],
          "labels": {
            "app.kubernetes.io/name": "hello-kubernetes",
          },
          "name": "hello-kubernetes",
          "namespace": undefined,
          "ownerReferences": [],
        },
        "name": "hello-kubernetes",
        "node": Node {
          "_children": {},
          "_context": {},
          "_dependencies": Set {},
          "_locked": false,
          "_metadata": [],
          "_validations": [],
          "host": [Circular],
          "id": "deployment",
          "scope": [Circular],
        },
        "patches": [],
        "props": {
          "apiVersion": "apps/v1",
          "kind": "Deployment",
          "metadata": {
            "labels": {
              "app.kubernetes.io/name": "hello-kubernetes",
            },
            "name": "hello-kubernetes",
          },
          "spec": {
            "replicas": 1,
            "selector": {
              "matchLabels": {
                "app.kubernetes.io/name": "hello-kubernetes",
              },
            },
            "template": {
              "metadata": {
                "labels": {
                  "app.kubernetes.io/name": "hello-kubernetes",
                },
                "name": "hello-kubernetes",
              },
              "spec": {
                "containers": [
                  {
                    "env": [
                      {
                        "name": "PORT",
                        "value": "8080",
                      },
                      {
                        "name": "MESSAGE",
                        "value": "Message goes here",
                      },
                      {
                        "name": "KUBERNETES_NAMESPACE",
                        "valueFrom": {
                          "fieldRef": {
                            "fieldPath": "metadata.namespace",
                          },
                        },
                      },
                      {
                        "name": "KUBERNETES_POD_NAME",
                        "valueFrom": {
                          "fieldRef": {
                            "fieldPath": "metadata.name",
                          },
                        },
                      },
                      {
                        "name": "KUBERNETES_NODE_NAME",
                        "valueFrom": {
                          "fieldRef": {
                            "fieldPath": "spec.nodeName",
                          },
                        },
                      },
                      {
                        "name": "CONTAINER_IMAGE",
                        "value": "paulbouwer/hello-kubernetes:1.10.1",
                      },
                    ],
                    "image": "paulbouwer/hello-kubernetes:1.10.1",
                    "imagePullPolicy": "IfNotPresent",
                    "livenessProbe": {
                      "httpGet": {
                        "port": IntOrString {
                          "value": 8080,
                        },
                      },
                    },
                    "name": "hello-kubernetes",
                    "ports": [
                      {
                        "containerPort": 8080,
                        "name": "http",
                        "protocol": "TCP",
                      },
                    ],
                    "readinessProbe": {
                      "httpGet": {
                        "port": IntOrString {
                          "value": 8080,
                        },
                      },
                    },
                  },
                ],
                "serviceAccountName": "hello-kubernetes",
              },
            },
          },
        },
        Symbol(@aws-cdk/core.DependableTrait): {
          "dependencyRoots": [
            [Circular],
          ],
        },
      },
      "ingress": KubeIngress {
        "apiGroup": "networking.k8s.io",
        "apiVersion": "networking.k8s.io/v1",
        "chart": [Circular],
        "kind": "Ingress",
        "metadata": ApiObjectMetadataDefinition {
          "_additionalAttributes": {
            "annotations": {
              "kubernetes.io/ingress.class": "traefik",
              "traefik.ingress.kubernetes.io/router.tls": "false",
            },
            "labels": {
              "app.kubernetes.io/name": "hello-kubernetes",
            },
            "name": "hello-kubernetes",
            "namespace": undefined,
          },
          "annotations": {
            "kubernetes.io/ingress.class": "traefik",
            "traefik.ingress.kubernetes.io/router.tls": "false",
          },
          "finalizers": [],
          "labels": {
            "app.kubernetes.io/name": "hello-kubernetes",
          },
          "name": "hello-kubernetes",
          "namespace": undefined,
          "ownerReferences": [],
        },
        "name": "hello-kubernetes",
        "node": Node {
          "_children": {},
          "_context": {},
          "_dependencies": Set {},
          "_locked": false,
          "_metadata": [],
          "_validations": [],
          "host": [Circular],
          "id": "ingress",
          "scope": [Circular],
        },
        "patches": [],
        "props": {
          "apiVersion": "networking.k8s.io/v1",
          "kind": "Ingress",
          "metadata": {
            "annotations": {
              "kubernetes.io/ingress.class": "traefik",
              "traefik.ingress.kubernetes.io/router.tls": "false",
            },
            "labels": {
              "app.kubernetes.io/name": "hello-kubernetes",
            },
            "name": "hello-kubernetes",
          },
          "spec": {
            "rules": [
              {
                "host": "test.h6020-001.devops-at-ho.me",
                "http": {
                  "paths": [
                    {
                      "backend": {
                        "service": {
                          "name": "hello-kubernetes",
                          "port": {
                            "number": 8080,
                          },
                        },
                      },
                      "path": "/",
                      "pathType": "Prefix",
                    },
                  ],
                },
              },
            ],
          },
        },
        Symbol(@aws-cdk/core.DependableTrait): {
          "dependencyRoots": [
            [Circular],
          ],
        },
      },
      "service": KubeService {
        "apiGroup": "core",
        "apiVersion": "v1",
        "chart": [Circular],
        "kind": "Service",
        "metadata": ApiObjectMetadataDefinition {
          "_additionalAttributes": {
            "labels": {
              "app.kubernetes.io/name": "hello-kubernetes",
            },
            "name": "hello-kubernetes",
            "namespace": undefined,
          },
          "annotations": {},
          "finalizers": [],
          "labels": {
            "app.kubernetes.io/name": "hello-kubernetes",
          },
          "name": "hello-kubernetes",
          "namespace": undefined,
          "ownerReferences": [],
        },
        "name": "hello-kubernetes",
        "node": Node {
          "_children": {},
          "_context": {},
          "_dependencies": Set {},
          "_locked": false,
          "_metadata": [],
          "_validations": [],
          "host": [Circular],
          "id": "service",
          "scope": [Circular],
        },
        "patches": [],
        "props": {
          "apiVersion": "v1",
          "kind": "Service",
          "metadata": {
            "labels": {
              "app.kubernetes.io/name": "hello-kubernetes",
            },
            "name": "hello-kubernetes",
          },
          "spec": {
            "ports": [
              {
                "name": "http",
                "port": 8080,
              },
            ],
            "selector": {
              "app.kubernetes.io/name": "hello-kubernetes",
            },
            "type": "LoadBalancer",
          },
        },
        Symbol(@aws-cdk/core.DependableTrait): {
          "dependencyRoots": [
            [Circular],
          ],
        },
      },
      "serviceAccount": KubeServiceAccount {
        "apiGroup": "core",
        "apiVersion": "v1",
        "chart": [Circular],
        "kind": "ServiceAccount",
        "metadata": ApiObjectMetadataDefinition {
          "_additionalAttributes": {
            "labels": {
              "app.kubernetes.io/name": "hello-kubernetes",
            },
            "name": "hello-kubernetes",
            "namespace": undefined,
          },
          "annotations": {},
          "finalizers": [],
          "labels": {
            "app.kubernetes.io/name": "hello-kubernetes",
          },
          "name": "hello-kubernetes",
          "namespace": undefined,
          "ownerReferences": [],
        },
        "name": "hello-kubernetes",
        "node": Node {
          "_children": {},
          "_context": {},
          "_dependencies": Set {},
          "_locked": false,
          "_metadata": [],
          "_validations": [],
          "host": [Circular],
          "id": "serviceAccount",
          "scope": [Circular],
        },
        "patches": [],
        "props": {
          "apiVersion": "v1",
          "kind": "ServiceAccount",
          "metadata": {
            "labels": {
              "app.kubernetes.io/name": "hello-kubernetes",
            },
            "name": "hello-kubernetes",
          },
        },
        Symbol(@aws-cdk/core.DependableTrait): {
          "dependencyRoots": [
            [Circular],
          ],
        },
      },
    },
    "_context": {},
    "_dependencies": Set {},
    "_locked": false,
    "_metadata": [],
    "_validations": [],
    "host": [Circular],
    "id": "HelloKubernetes",
    "scope": App {
      "node": Node {
        "_children": {
          "HelloKubernetes": [Circular],
        },
        "_context": {},
        "_dependencies": Set {},
        "_locked": false,
        "_metadata": [],
        "_validations": [],
        "host": [Circular],
        "id": "",
        "scope": undefined,
      },
      "outdir": "dist",
      "outputFileExtension": ".k8s.yaml",
      "recordConstructMetadata": false,
      "yamlOutputType": 1,
      Symbol(@aws-cdk/core.DependableTrait): {
        "dependencyRoots": [
          [Circular],
        ],
      },
    },
  },
  Symbol(@aws-cdk/core.DependableTrait): {
    "dependencyRoots": [
      [Circular],
    ],
  },
}
`;
